sqlbrite-contentprovider/src/main/java/com/squareup/sqlbrite_contentprovider/SqlBriteContentProvider.java

```diff
@@ -0,0 +1,230 @@
+          Log.v("SqlBrite", message);
+        }
+      };
+    }
+    logging = enabled;
+  }
+
+  /**
+   * Specify a custom logger for debug messages when {@linkplain #setLoggingEnabled(boolean)
+   * logging is enabled}.
+   */
+  public void setLogger(Logger logger) {
+    if (logger == null) throw new NullPointerException("logger == null");
+    this.logger = logger;
+  }
+
+  /**
+   * Create an observable which will notify subscribers with a {@linkplain Query query} for
+   * execution. Subscribers are responsible for <b>always</b> closing {@link Cursor} instance
+   * returned from the {@link Query}.
+   * <p>
+   * Subscribers will receive an immediate notification for initial data as well as subsequent
+   * notifications for when the supplied {@code uri}'s data changes. Unsubscribe when you no longer
+   * want updates to a query.
+   * <p>
+   * <b>Warning:</b> this method does not perform the query! Only by subscribing to the returned
+   * {@link Observable} will the operation occur.
+   *
+   * @see ContentResolver#query(Uri, String[], String, String[], String)
+   * @see ContentResolver#registerContentObserver(Uri, boolean, ContentObserver)
+   */
+  public Observable<Query> createQuery(@NonNull final Uri uri, @Nullable final String[] projection,
+      @Nullable final String selection, @Nullable final String[] selectionArgs, @Nullable
+  final String sortOrder, final boolean notifyForDescendents) {
+    final Query query = new Query() {
+      @Override public Cursor run() {
+        return query(uri, projection, selection, selectionArgs, sortOrder);
+      }
+    };
+    return Observable.create(new Observable.OnSubscribe<Query>() {
+      @Override public void call(final Subscriber<? super Query> subscriber) {
+        contentResolver.registerContentObserver(uri, notifyForDescendents,
+            new ContentObserver(contentObserverHandler) {
+              @Override public void onChange(boolean selfChange) {
+                if (subscriber.isUnsubscribed()) {
+                  contentResolver.unregisterContentObserver(this);
+                } else {
+                  subscriber.onNext(query);
+                }
+              }
+            });
+      }
+    }).startWith(query);
+  }
+
+  /**
+   * Query the given URI, returning a {@link Cursor} over the result set.
+   * <p>
+   * For best performance, the caller should follow these guidelines:
+   * <ul>
+   * <li>Provide an explicit projection, to prevent
+   * reading data from storage that aren't going to be used.</li>
+   * <li>Use question mark parameter markers such as 'phone=?' instead of
+   * explicit values in the {@code selection} parameter, so that queries
+   * that differ only by those values will be recognized as the same
+   * for caching purposes.</li>
+   * </ul>
+   * </p>
+   *
+   * @see ContentResolver#query(Uri, String[], String, String[], String)
+   */
+  public final Cursor query(@NonNull Uri uri, @Nullable String[] projection,
```

Comment:
The class is `final` so you can remove it from all these methods. Really need to get checkstyle all up in here...